### ISSUE TITLE
Show/hide model toggle under "Edit Display Elements" is inoperable on MacOS Sequoia

### ISSUE THREAD

**Issue by nkaminski** [xLightsSequencer/xLights#5015]
**Describe the bug**

**To Reproduce**
1. In the sequencer tab, right click on a model and choose "edit display elements" option
2. Click on the eye icon to try to hide the model
3. Observe that it is impossible to hide the model
4. Right click on one model and choose "hide all"
5. Observe that is is now impossible to selectively unhide any of the models

**Expected behavior**
Single clicking the eye icon for a model in the display elements pane should toggle the visibility of that model 

**Screenshots**
Video of the issue attached

**Versions (please complete the following information):**
 - OS: MacOS Sequoia 15.1.1 (24B91)
 - xLights version 2024.19

https://github.com/user-attachments/assets/7cb8d1a0-650a-412b-9b71-b21b0e8858a6

**Comment by nkaminski** [comment]
I can also confirm that this behavior does **not** happen on Windows and is _exclusive_ to the Mac OS build of xlights.

**Comment by cybercop23** [comment]
Can't duplicate. Tried my existing show folder and a fresh folder with only a Master View (like you have) and once I click the eye-ball, it hides it, I can unhide it, hide all show all.

**Comment by nkaminski** [comment]
Will try as a separate macOS user to see if this is related to a specific setting or the like later today.

**Comment by derwin12** [comment]
Tools->Package logs might help us figure this out..

**Comment by nkaminski** [comment]
Can confirm that the behavior persists when tested using a separate show directory as well as a separate MacOS user. @cybercop23  can you confirm that you are using MacOS Sequoia 15.1.1 (24B91) on arm64 (apple silicon) as your platform?

Logs after reproducing this with a dedicated show directory and empty sequence containing only 2 models:
[xLightsProblem.zip](https://github.com/user-attachments/files/17986429/xLightsProblem.zip)

**Comment by nkaminski** [comment]
I also opted to experiment with rolling back to earlier versions of xLights obtained from https://dankulp.com/xlights/archive/  as I recall this function working as it should in the past and have made the following observations:

* [xLights-2023.18](https://dankulp.com/xlights/archive/xLights-2023.18.dmg) - working
* [xLights-2023.22.1](https://dankulp.com/xlights/archive/xLights-2023.22.1.dmg) - working
* [xLights-2024.02.1.dmg](https://dankulp.com/xlights/archive/xLights-2024.02.1.dmg) - NOT working

Therefore the cause of this behavior seems to be something that was introduced between releases 2023.22.1 and 2024.02.1.

**Comment by cybercop23** [comment]
@nkaminski yes, I'm a Sequoia, M2 using the latest TestFlight version.

**Comment by cybercop23** [comment]
Yes... I get interim updates with each commit. But nothing that would have broken or fixed this.

**Comment by nkaminski** [comment]
So another interesting datapoint: I have built the latest master branch of xLights from source and still do not have an operable show/hide display elements option.

Therefore, I tried setting a breakpoint at https://github.com/xLightsSequencer/xLights/blob/master/xLights/ViewsModelsPanel.cpp#L993 and it appears that I _never hit this breakpoint_ therefore are we certain that the TestFlight version that you are using has been built against the exact same version and patchlevel of wxWidgets as version 2024.19?

**Comment by nkaminski** [comment]
OK, so after some analysis in the Xcode debugger, I finally understand in detail what's going on here:

The root of the issue appears to be the icon rect/bounding box size computation in wxWidgets at src/generic/listctrl.cpp:1750 computing a bounding box that is half the actual with an height of the icon on HiDPI displays.

Therefore, one can work around the issue by clicking within 8 pixels of the top left corner of the row containing the eye icon versus anywhere on the icon (highlighted in red in my image).
<img width="209" alt="rect size bug" src="https://github.com/user-attachments/assets/9f2b9ca4-5153-4654-8aca-bef3a12e1c14">

**Comment by nkaminski** [comment]
Can reproduce the behavior on:

* Internal display of my M3 MacBook Pro 16in
* External LG Ultrafine 5K display
* Internal display of the 2019 Intel MacBook Pro

All of these are ~200ppi displays, for which macOS is rendering at 2x scale.

What I think may be happening is wxWidgets is computing the bounding box in function HitTest() as icon_image_dimension / scale_factor instead of icon_image_dimension * scale_factor; as in its _shrinking_ the rect used to test if a click is on the icon as opposed to _growing_ it (as the icon image is 2x as many pixels wide and tall on HiDPI vs standard).
